meson.build

```diff
@@ -339,16 +339,16 @@ basic_disabled_warnings = [
 possible_common_cc_flags = [
         '-Wdate-time',
         '-Wendif-labels',
-        '-Werror=format=2',
-        '-Werror=implicit-function-declaration',
-        '-Werror=incompatible-pointer-types',
-        '-Werror=int-conversion',
-        '-Werror=overflow',
-        '-Werror=return-type',
-        '-Werror=shift-count-overflow',
-        '-Werror=shift-overflow=2',
-        '-Werror=undef',
-        '-Werror=unused-function',
+        '-Wformat=2',
+        '-Wimplicit-function-declaration',
+        '-Wincompatible-pointer-types',
+        '-Wint-conversion',
+        '-Woverflow',
+        '-Wreturn-type',
+        '-Wshift-count-overflow',
+        '-Wshift-overflow=2',
+        '-Wundef',
+        '-Wunused-function',
```

Comment:
I think it's OK to drop `-Werror' for unused-function: after all, because the code will work fine in that condition, it's just that there's some cleanup/refactoring that is unfinished. But the other conditions are actual errors: we never want to run code for which they are true. And even when doing development, it's just better to fail early in those conditions. In particular, warnings can be easily missed when doing scripted builds.

Thus, please drop everything except this line.